src/lib/search-select.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  createPrompt,
+  useState,
+  useKeypress,
+  usePrefix,
+  useRef,
+  isEnterKey,
+  isBackspaceKey,
+  KeypressEvent,
+  type PromptConfig,
+} from "@inquirer/core";
+import type {} from "@inquirer/type";
+import chalk from "chalk";
+import figures from "figures";
+
+export type Choice = {
+  value: string;
+};
+
+export type SelectConfig = PromptConfig<{
+  choices: ReadonlyArray<Choice>;
+  pageSize?: number;
+}>;
+
+function renderItem({ item, isActive }: { item: Choice; isActive: boolean }) {
+  const line = item.value;
+
+  const prefix = isActive ? figures.pointer : ` `;
+  const str = `${prefix} ${line}`;
+  return isActive ? chalk.cyan(str) : str;
+}
+
+// @inquirer/core has these, but it includes the vim keybinds j and k, which
+// conflict with typing those letters.
+const isUpKey = (key: KeypressEvent): boolean =>
+  key.name === "up" || // The up key
+  (key.ctrl && key.name === "p"); // Emacs keybinding
+
+const isDownKey = (key: KeypressEvent): boolean =>
+  key.name === "down" || // The down key
+  (key.ctrl && key.name === "n"); // Emacs keybinding
+
+export const searchSelect = createPrompt(
+  (config: SelectConfig, done: (value: string) => void) => {
+    const { choices } = config;
+    const pageSize = config.pageSize ?? 7;
+
+    const firstRender = useRef(true);
+    const prefix = usePrefix();
+    const [status, setStatus] = useState<"pending" | "done">("pending");
+    const [typed, setTyped] = useState("");
+
+    // TODO: Fuzzy filter?
+    const items = choices.filter((item, _) => item.value.startsWith(typed));
+
+    const [activeState, setActive] = useState(0);
+    const [scrollState, setScroll] = useState(0);
```

Comment:
on first read I'm struggling to understand what scroll is tracking, could you add a comment explaining that at a high level?